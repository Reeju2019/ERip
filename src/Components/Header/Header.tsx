import {
    Button,
    Container,
    Image,
    Modal,
    Row,
    InputGroup,
    Form,
    Nav,
} from 'react-bootstrap';
import { useState } from 'react';
import './Header.css';
const Header: React.FunctionComponent = () => {
    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    return (
        <>
            <Container className="mt-2">
                <Container fluid>
                    <Row className="Navbar">
                        <Image
                            src="/static/image/logo_erip.png"
                            className="col-2 Logo align-self-center"
                            alt="logo"
                        />

                        <div className="modal-search d-flex col-6">
                            {/* Modal */}
                            <Button
                                type="button"
                                className="btn modal-btn rounded-0 bg-transparent text-muted fw-bold"
                                data-toggle="modal"
                                data-target="#myModal"
                                onClick={handleShow}
                            >
                                <i className="fas fa-map-marker-alt"></i>
                                Bengaluru
                                <i className="fa-solid fa-caret-down"></i>
                            </Button>
                            <Modal
                                show={show}
                                onHide={handleClose}
                                backdrop="static"
                                keyboard={false}
                            >
                                <Modal.Header closeButton>
                                    <Modal.Title>Current Location</Modal.Title>
                                </Modal.Header>
                                <Modal.Body>have to implement</Modal.Body>
                                <Modal.Footer>
                                    <Button
                                        variant="secondary"
                                        onClick={handleClose}
                                    >
                                        Close
                                    </Button>
                                    <Button variant="primary">
                                        Understood
                                    </Button>
                                </Modal.Footer>
                            </Modal>

                            {/* <!--End of Modal -->  */}

                            <InputGroup className="search_btn d-flex">
                                <div className="align-self-center m-3">
                                    <i className="fas fa-search"></i>
                                </div>
                                <Form.Control
                                    type="search"
                                    className="form-control rounded-0 border-0"
                                    id="search-border"
                                    placeholder="Search for a services"
                                    aria-label="Search"
                                    aria-describedby="search-addon"
                                />
                            </InputGroup>
                        </div>
                        <Nav className="modal-search d-flex col-4 justify-content-end align-self-center">
                            <Nav.Link
                                href="home"
                                className="text-muted fw-bold"
                            >
                                Home
                            </Nav.Link>
                            <Nav.Link
                                href="contact"
                                className="text-muted fw-bold"
                            >
                                Contact Us
                            </Nav.Link>
                            <Nav.Link
                                href="/login"
                                className="d-flex flex-row text-muted fw-bold"
                            >
                                <i className="fa-regular fa-user"></i>
                                Login
                            </Nav.Link>
                        </Nav>
                    </Row>
                </Container>
            </Container>
        </>
    );
};

export default Header;
