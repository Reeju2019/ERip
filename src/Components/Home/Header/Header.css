.header-comp {
  position: relative;
}

.fa-map-marker-alt {
  margin-right: 9%;
}

.fa-regular {
  margin-right: 9%;
  margin-top: 8%;
}

.fa-solid {
  margin-left: 2%;
}

.input-group {
  width: 100%;
}
.input-group-text {
  background-color: white;
  border-radius: 0%;
  width: 100%;
  border: 1px solid #8080808a;
}

#search-border:focus {
  outline: none !important;
}

.Navbar {
  margin: 0 12%;
}

.nav-link {
  color: black;
}

.nav-link:hover {
  color: rgb(23 104 205);
}

.Logo {
  /* border: 2px solid red; */
  height: 30px;
  width: 80px !important;
}

.modal-btn {
  border: 1px solid #8080808a !important;
  width: 40%;
}
.search_btn {
  border: 1px solid #8080808a !important;
  /* border-left: 0!important; */
}

.btn-close {
  width: 5px !important;
}
.caro-img {
  width: 25vw;
  margin-right: 6vmax;
}

.gps {
  cursor: pointer;
  flex-direction: row;
}
.carousel-container-1 {
  background-size: cover;
  background-position: center;
}
#log_modal .modal-header{
  border: none !important;
  text-align: center !important;
}
#log_modal span{
  font-weight:normal !important;
  color: #80808099;
  font-size: 20px;
}
#log_modal .btn-close{
  float: inline-start !important;
}
