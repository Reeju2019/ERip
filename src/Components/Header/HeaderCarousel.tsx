import * as React from 'react';
import { Carousel, Container, Image } from 'react-bootstrap';
import './Header.css';

interface IHeaderCarouselProps {}

const HeaderCarousel: React.FunctionComponent<IHeaderCarouselProps> = (
    props
) => {
    return (
        <>
            <Container fluid className="carousel-container-1 mt-2">
                <div className="row carousel-content mx-4">
                    <div
                        id="carouselExampleIndicators"
                        className="carousel slide"
                        data-bs-ride="true"
                    >
                        <Carousel>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Repair & Maintainence <br />
                                            Instant Doorstep Services On Demand
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Top Trained Experts <br/> at Your Doorstep
                                    Right Away!
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                            <Carousel.Item>
                                <div className="d-flex flex-row">
                                    <div className="w-100 d-flex justify-content-center align-item-center align-self-center">
                                        <h2 className="txt col-8 justify-content-start text-white fs-2">
                                            Easy Fix <br/> Superior Service Quality Parts
                                        </h2>
                                    </div>
                                    <Image
                                        className="d-flex caro-img justify-content-center"
                                        src="/static/image/carousel-img-1.png"
                                        alt="First slide"
                                    />
                                </div>
                            </Carousel.Item>
                        </Carousel>
                    </div>
                </div>
            </Container>
        </>
    );
};

export default HeaderCarousel;
