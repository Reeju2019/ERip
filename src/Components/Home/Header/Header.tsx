import { Button, Container, Image, Modal, Row, InputGroup, Form, Nav } from 'react-bootstrap'
import { useState } from 'react'
import './Header.css'
import HeaderConstant from '../../../Data/HeaderConstant.mock.json'
import Geocode from 'react-geocode'
import Key from '../../../key.json'

const Header: React.FunctionComponent = () => {
  const [city, setCity] = useState('Bengaluru')
  const [error, setError] = useState()
  // GPS
  const geoApi: string = Key.GOOGLE_MAP_API_KEY
  Geocode.setApiKey(geoApi)
  Geocode.setLanguage('en')
  Geocode.setRegion('es')
  Geocode.setLocationType('ROOFTOP')
  Geocode.enableDebug()

  const getLocation = () => {
    navigator.geolocation.getCurrentPosition(function (pos) {
      const lat = pos.coords.latitude.toString()
      const long = pos.coords.longitude.toString()
      Geocode.fromLatLng(lat, long).then(
        (response) => {
          for (let i = 0; i < response.results[0].address_components.length; i++) {
            for (let j = 0; j < response.results[0].address_components[i].types.length; j++) {
              if (
                response.results[0].address_components[i].types[j] ===
                response.results[0].address_components[i].long_name
              ) {
                setCity(response.results[0].address_components[i].long_name)
              }
            }
          }
        },
        (error) => {
          setError(error.message)
        },
      )
    })
  }
  // end GPS
  const [show, setShow] = useState(false)

  const handleClose = () => setShow(false)
  const handleShow = () => setShow(true)
  return (
    <>
      <Container className='mt-2 header-comp'>
        <Container fluid>
          <Row className='Navbar'>
            <Image
              src='/static/image/logo_erip.png'
              className='col-2 Logo align-self-center'
              alt='logo'
            />

            <div className='modal-search d-flex col-6'>
              {/* Modal */}
              <Button
                type='button'
                className='btn modal-btn rounded-0 bg-transparent text-muted fw-bold'
                data-toggle='modal'
                data-target='#myModal'
                onClick={handleShow}
              >
                <i className='fas fa-map-marker-alt' />
                {city}
                <i className='fa-solid fa-caret-down' />
              </Button>
              <Modal show={show} onHide={handleClose} backdrop='static' keyboard={true}>
                <Modal.Header className='d-flex m-1'>
                  <div className='col-5'>{HeaderConstant.location}</div>
                  <div className='col-7 d-flex justify-content-end gps' onClick={getLocation}>
                    {HeaderConstant.gps}
                    <i className='fa-solid fa-angle-right align-self-center' />
                  </div>
                </Modal.Header>
                <Modal.Body>
                  <InputGroup className='d-flex search_btn'>
                    <div className=' lign-self-center m-3'>
                      <i className='fas fa-search' />
                    </div>
                    <Form.Control
                      type='search'
                      className='form-control rounded-0 border-0'
                      id='search-border'
                      placeholder='Search for a services'
                      aria-label='Search'
                      aria-describedby='search-addon'
                    />
                  </InputGroup>
                </Modal.Body>
                <Modal.Footer>{error}</Modal.Footer>
              </Modal>

              {/* <!--End of Modal -->  */}

              <InputGroup className='search_btn d-flex'>
                <div className='align-self-center m-3'>
                  <i className='fas fa-search' />
                </div>
                <Form.Control
                  type='search'
                  className='form-control rounded-0 border-0'
                  id='search-border'
                  placeholder='Search for a services'
                  aria-label='Search'
                  aria-describedby='search-addon'
                />
              </InputGroup>
            </div>
            <Nav className='modal-search d-flex col-4 justify-content-end align-self-center'>
              <Nav.Link href='/' className='text-muted fw-bold'>
                {HeaderConstant.home}
              </Nav.Link>
              <Nav.Link href='/contact' className='text-muted fw-bold'>
                {HeaderConstant.contact_us}
              </Nav.Link>
              <Nav.Link href='/login' className='d-flex flex-row text-muted fw-bold'>
                <i className='fa-regular fa-user' />
                {HeaderConstant.login}
              </Nav.Link>
            </Nav>
          </Row>
        </Container>
      </Container>
    </>
  )
}

export default Header
