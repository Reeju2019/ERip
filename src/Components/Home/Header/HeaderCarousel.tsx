import { Carousel, Container, Image } from 'react-bootstrap'
import './Header.css'
import HeaderData from '../../../Data/Header.mock.json'

const HeaderCarousel: React.FunctionComponent = () => {
  return (
    <>
      <Container fluid className='carousel-container-1 mt-2'>
        <div className='row carousel-content mx-4'>
          <div id='carouselExampleIndicators' className='carousel slide' data-bs-ride='true'>
            <Carousel>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_1_1}
                      <br />
                      {HeaderData.header_caro_1_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-1.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_2_1} <br />
                      {HeaderData.header_caro_2_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-2.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
              <Carousel.Item>
                <div className='d-flex flex-row'>
                  <div className='w-100 d-flex justify-content-center align-item-center align-self-center'>
                    <h2 className='txt col-8 justify-content-start text-white fs-2'>
                      {HeaderData.header_caro_3_1} <br />
                      {HeaderData.header_caro_3_2}
                    </h2>
                  </div>
                  <Image
                    className='d-flex caro-img justify-content-center'
                    src='/static/image/carousel-img-3.png'
                    alt='First slide'
                  />
                </div>
              </Carousel.Item>
            </Carousel>
          </div>
        </div>
      </Container>
    </>
  )
}

export default HeaderCarousel
