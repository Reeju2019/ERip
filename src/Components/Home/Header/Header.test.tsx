import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Header from './Header'
import { MemoryRouter as Router } from 'react-router-dom'

test('header', () => {
  render(
    <Router>
      <Header />
    </Router>,
  )
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(
      <Router>
        <Header />
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(
      <Router>
        <Header />
      </Router>,
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
