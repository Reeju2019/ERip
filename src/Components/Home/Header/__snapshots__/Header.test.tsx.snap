// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
<div
  className=" py-1 m-0 header-comp container-fluid"
>
  <div
    className="container"
  >
    <div
      className="Navbar row"
    >
      <div
        className="col-1 align-item-center align-self-center col"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="col-2 Logo align-self-center"
            src="/static/image/logo_erip.png"
          />
        </a>
      </div>
      <div
        className="modal-search d-flex col-6 col"
      >
        <button
          className="btn modal-btn rounded-0 bg-transparent text-muted fw-bold btn btn-primary"
          data-target="#myModal"
          data-toggle="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-map-marker-alt"
          />
          Bengaluru
          <i
            className="fa-solid fa-caret-down"
          />
        </button>
        <div
          className="search_btn d-flex input-group"
        >
          <div
            className="align-self-center m-3"
          >
            <i
              className="fas fa-search"
            />
          </div>
          <input
            aria-describedby="search-addon"
            aria-label="Search"
            className="form-control rounded-0 border-0 form-control"
            id="search-border"
            placeholder="Search for a services"
            type="search"
          />
        </div>
      </div>
      <div
        className="col-5 align-self-center col"
      >
        <div
          className="modal-search d-flex justify-content-end align-self-center nav"
          onKeyDown={[Function]}
        >
          <a
            className="text-muted fw-bold nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="text-muted fw-bold nav-link"
            data-rr-ui-event-key="/contact"
            disabled={false}
            href="/contact"
            onClick={[Function]}
          >
            Contact Us
          </a>
          <a
            className="d-flex flex-row text-muted fw-bold nav-link"
            data-rr-ui-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <i
              className="fa-regular fa-user"
            />
            Login
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
